src/p5.dance.js

```diff
@@ -191,35 +193,18 @@ module.exports = class DanceParty {
   }
 
   preload() {
-    // Load spritesheets compressed to various levels of quality with pngquant
-    // Pass queryparam ?quality=<quality> to try a particular quality level.
-    // Only those png assets will be downlaoded.
-    // Available quality levels:
-    // 50 - 40% smaller
-    // 25 - 46%
-    // 10 - 51%
-    //  5 - 55%
-    //  1 - 55%
-    //  0 - 63% smaller
-    let qualitySuffix = '-q50'; // Default to q50 for now.  Set to '' to go back to full-quality.
-    const qualitySetting = queryParam('quality');
-    if (qualitySetting) {
-      qualitySuffix = `-q${qualitySetting}`;
-      document.title = `q${qualitySetting} - ${document.title}`;
-    }
-
-    // Load spritesheet JSON files
-    this.world.SPRITE_NAMES.forEach(this_sprite => {
-      ANIMATIONS[this_sprite] = [];
-      this.world.MOVE_NAMES.forEach(({ name, mirror }, moveIndex) => {
-        const baseUrl = `${this.assetBase}${this_sprite}_${name}`;
-        this.p5_.loadJSON(`${baseUrl}.json`, jsonData => {
-          // Passing true as the 3rd arg to loadSpriteSheet() indicates that we want
-          // it to load the image as a Image (instead of a p5.Image), which avoids
-          // a canvas creation. This makes it possible to run on mobile Safari in
-          // iOS 12 with canvas memory limits.
-          this.setAnimationSpriteSheet(this_sprite, moveIndex,
-            this.p5_.loadSpriteSheet(`${baseUrl}${qualitySuffix}.png`, jsonData.frames, true), mirror)
+    this.resourceLoader_.getAnimationData(animationData => {
+      this.world.SPRITE_NAMES.forEach(costume => {
+        const costumeData = animationData[costume.toLowerCase()];
+        ANIMATIONS[costume] = [];
+        this.world.MOVE_NAMES.forEach(({ name: moveName, mirror }, moveIndex) => {
+          const moveData = costumeData[moveName.toLowerCase()];
+          this.setAnimationSpriteSheet(
+            costume,
+            moveIndex,
+            this.resourceLoader_.loadSpriteSheet(moveData.spritesheet, moveData.frames),
```

Comment:
so we still call `loadSpriteSheet()` once per character per move, it's just that we refer to the same spritesheet URL in some places, so the "load" is instantaneous, correct? We will create multiple `Image` elements that refer to the same spritesheet, but hopefully the browsers are all smart enough to realize that they only need one underlying instance. If we were using `p5.Image` elements, I think this approach would be a little more dangerous. We may end up with multiple canvases with the exact same spritesheet in each.

Another approach (that may be required, but I'm not certain) would be to break the `Image` loading from the `SpriteSheet` loading. You'd first call our new `p5.loadImageElement()` with each unique URL (4-5 per character), then you can call `loadSpriteSheet()` (which would then no longer be async) for each dance, passing the loaded `Image` in as the 1st param (instead of the `url`)